packages/sdk-browser/src/sdk-client.ts

```diff
@@ -246,8 +252,145 @@ export class TurnkeyPasskeyClient extends TurnkeyBrowserClient {
+    // Convert exported private key to Uint8Array
+    const rawPrivateKeyBytesPKCS8 = new Uint8Array(exportedPrivateKey);
+
+    // PKCS#8 private key is structured with the key data at a specific position
+    // The actual key starts at byte 36 and is 32 bytes long
+    const rawTurnkeyApiPrivateKey = rawPrivateKeyBytesPKCS8.slice(36, 36 + 32);
+
+    const exportedPublicKey = await crypto.subtle.exportKey(
+      "raw",
+      p256key.publicKey,
+    );
+  
+    const compressedPublicKey = compressPubKey(new Uint8Array(exportedPublicKey));
+
+    // Step 3: add API key to Turnkey User
+    await this.createApiKeys({
+      apiKeys: [{
+        apiKeyName: "Session Key",
+        publicKey: uint8ArrayToHexString(compressedPublicKey),
+        expirationSeconds: expirationSeconds ?? "900", // default to 15 minutes
+      }],
+      userId,
+    });
+
+    // Step 4: set up encryption
+    const suite = new CipherSuite({
+      kem: KemId.DhkemP256HkdfSha256,
+      kdf: KdfId.HkdfSha256,
+      aead: AeadId.Aes256Gcm,
+    });
+
+    // Step 5: import the target embedded key (from the iframe)
+    const targetKeyBytes = uint8ArrayFromHexString(targetEmbeddedKey);
+    const targetKey = await crypto.subtle.importKey(
+      "raw",
+      targetKeyBytes,
+      {
+        name: "ECDH",
+        namedCurve: "P-256",
+      },
+      true,
+      []
+    );
+
+    // Step 6: sender encrypts a message with the recipient public key.
+    const sender = await suite.createSenderContext({
+      recipientPublicKey: targetKey,
+      info: TURNKEY_HPKE_INFO,
+    });
+    const ciphertext = await sender.seal(
+      rawTurnkeyApiPrivateKey,
+      additionalAssociatedData(new Uint8Array(sender.enc), targetKeyBytes)
+    );
+    const ciphertextUint8Array = new Uint8Array(ciphertext);
+
+    // Step 7: assemble bundle
+    const encappedKey = new Uint8Array(sender.enc);
+    const compressedEncappedKey = compressPubKey(encappedKey);
+    const result = new Uint8Array(
+      compressedEncappedKey.length + ciphertextUint8Array.length
+    );
+    result.set(compressedEncappedKey);
+    result.set(ciphertextUint8Array, compressedEncappedKey.length);
+
+    // The equivalent of the email auth bundle. Can store in local storage
+    const base58encodedBundle = bs58check.encode(result);
+
+    // Store auth bundle in local storage
+    await setStorageValue(StorageKeys.AuthBundle, base58encodedBundle);
+
+    return base58encodedBundle;
+  };
 }
 
+// NOTE: the below can be moved to utility packages
+/**
+ * Create additional associated data (AAD) for AES-GCM decryption.
+ */
+const additionalAssociatedData = (
+  senderPubBuf: Uint8Array,
+  receiverPubBuf: Uint8Array,
+): Uint8Array => {
+  return new Uint8Array([
+    ...Array.from(senderPubBuf),
+    ...Array.from(receiverPubBuf),
+  ]);
+};
+
+export const compressPubKey = (publicKey: Uint8Array): Uint8Array => {
```

Comment:
let's move this to @turnkey/crypto?

Reply:
yep, precisely the reason for the crypto PR! 😄